juju/juju
# Merge cross model relations feature branch

This PR merges the cross model relations feature branch. The vast majority of the work has been reviewed previously for landing into the feature branch.

The main commit to look at is:
https://github.com/juju/juju/commit/dbf2140fd85c2b6608fe16fd8dcc7bd76cb72b67

The above commit adds the cross model feature flag to hide the cross model relations features - CLI, facades, and database collections.

As a side effect, the work in the feature branch also fixes:
https://bugs.launchpad.net/juju/+bug/1621754

QA:
bootstrap and use a 2.0.0 GA client to deploy and relate services
bootstrap and ensure cross model stuff works with feature flag
